src/main/java/org/mule/extension/db/internal/source/RowListener.java

```diff
@@ -0,0 +1,219 @@
+   * processed twice by concurrent polls.
+   */
+  @Parameter
+  @Optional
+  private String idColumn;
+
+  @ParameterGroup(name = ADVANCED)
+  @NullSafe
+  private QuerySettings settings;
+
+  @Config
+  private DbConnector config;
+
+  @Connection
+  private ConnectionProvider<DbConnection> connectionProvider;
+
+  private final QueryResolver<ParameterizedStatementDefinition> queryResolver = new ParameterizedQueryResolver<>();
+  private ItemHandler idHandler;
+  private ItemHandler watermarkHandler;
+
+  @Override
+  protected void doStart() throws MuleException {
+    if (idColumn != null) {
+      idHandler = (item, row) -> {
+        Object id = row.get(idColumn);
+        if (id != null) {
+          item.setId(id.toString());
+        }
+      };
+    } else {
+      idHandler = new NullItemHandler();
+    }
+
+    if (watermarkColumn != null) {
+      watermarkHandler = (item, row) -> {
+        Object watermark = row.get(watermarkColumn);
+        if (watermark == null) {
+          if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug(format(
+                                "A null watermark value was obtained for row %s. Watermark value won't be updated for this row",
+                                row));
+          }
+
+          return;
+        }
+
+        if (!(watermark instanceof Serializable)) {
+          LOGGER.error(format("Watermark values need to be serializable, but a value of type %s was found instead for row %s",
+                              watermark.getClass().getName(), row));
+        }
+
+        item.setWatermark((Serializable) watermark);
+      };
+    } else {
+      watermarkHandler = new NullItemHandler();
+    }
+  }
+
+  @Override
+  protected void doStop() {
+
+  }
+
+  @Override
+  public void poll(PollContext<Map<String, Object>, Void> pollContext) {
+    if (pollContext.isSourceStopping()) {
+      return;
+    }
+
+    DbConnection connection;
+    try {
+      connection = connectionProvider.connect();
+    } catch (Exception e) {
+      if (e instanceof ConnectionException) {
+        pollContext.onConnectionException((ConnectionException) e);
+      }
+      LOGGER.error(format("Could not obtain connection while trying to poll table '%s'. %s", table, e.getMessage()), e);
+      return;
+    }
+
+    try {
+      QueryDefinition queryDefinition = new QueryDefinition();
+      StringBuilder sql = new StringBuilder("SELECT * FROM ").append(table);
+      pollContext.getWatermark().ifPresent(w -> {
+        sql.append(" WHERE ").append(watermarkColumn).append(" > :watermark");
+        queryDefinition.addInputParameter("watermark", w);
+      });
+
+      queryDefinition.setSql(sql.toString());
+      Query query = queryResolver.resolve(queryDefinition, config, connection, null);
+
+      QueryStatementFactory statementFactory = new QueryStatementFactory();
+      statementFactory.setFetchSize(settings.getFetchSize() != null ? settings.getFetchSize() : 0);
```

Comment:
why 0 and not 10 (the default for other cases)? in fact, why is not 10 the default of fetchSize?